libs/packages/components/src/lib/editor/editor.component.ts

```diff
@@ -117,4 +118,94 @@ export class SdsEditorComponent implements ControlValueAccessor {
     registerOnTouched(fn: any) {
         this._onTouched = fn;
     }
+
+    // onKeydown(event) {
+    //     if (event.key === 'Backspace' || event.key === 'Delete') {
+    //         this.length = this.length - 1;
+    //     }
+    // }
+
+    // onKeyPressed() {
+    //     this.length = this.length + 1;
+    // }
+
+    // Get the cursor current position and ma
+    getCaretCharacterOffsetWithin(element) {
+        let caretOffset = 0;
+        if (typeof window.getSelection != 'undefined') {
+            let range = window.getSelection().getRangeAt(0);
+            let preCaretRange = range.cloneRange();
+            preCaretRange.selectNodeContents(element);
+            preCaretRange.setEnd(range.endContainer, range.endOffset);
+            caretOffset = preCaretRange.toString().length;
+        } else if (
+            typeof this._document['selection'] != 'undefined' &&
+            this._document['selection'].type != 'Control'
```

Comment:
Can you add comment for this - in what situation would `window.getSelection` be undefined vs `document.selection`?

Reply:
This condition never happens, removed the code